### DISCUSSION TITLE
release cadence

### DISCUSSION THREAD

**Discussion by tessus** [dani-garcia/vaultwarden#2944] [General]
I am still a bit fuzzy on the release cadence, or better said when a release is cut.

The latest big change (group support) was merged over 3 weeks ago. Then a few days ago the changes were merged that should allow newer clients (2022.11.x) to use the vw server.

I am mainly asking, because my client just wanted to be restarted to update to the latest release, which of course I denied, because I doubt that the new login flow is supported by the current vw server.

Is there a matrix that shows which clients can be used with which version of vaultwarden? When can we expect a new vaultwarden release?
I am not saying a new release is absolutely necessary, I am just curious as to how the decision is made to create a new release. That's all.
Please don't take this the wrong way and read this post as "create a new release now". ;-) This is ddefinitely not how it was meant.

**Comment by BlackDex** [comment]
No prob. There isn't anything set in stone, like a x week release or something.

Depending on what we merge we sometimes want to take a bit more time on releasing a new version (in the hopes that there are people who using the `testing` branch releases) to filter out bugs or issues.

Like, currently, there is a bug in regards to the groups feature you mentioned, which has just been exposed, and in this case I'm trying to fix that, but encounter some items i really need to carefully check and test, which takes more time.

Overall, we at least try to update all the dependencies and do some releases in regards to that, even if there are no major new items added.

We also try to release a new version on major Bitwarden client releases, like the web-vault for example. But in this case we probably want to squash that groups bug, and after that release a new version i think.

**Reply by tessus** [reply]
Thanks for the info.

Proper testing is always a good idea and many software projects ignore that part way too often. Personally I rather wait for a release, if I can then be fairly sure that the release is stable.

The only problem I can see are new flows and API endpoints in the server. I am not entirely sure why Bitwarden opted to change the login flow. I majorly dislike one login screen to be split in 2 screens/steps. (Also not the best for security, when the server tells people that a user does not exist. Not saying that Bitwarden does that.) If one were to use 2FA, there'd be 3 separate screens and 3 times clicking on a button before being logged in. But that's just my personal opinion.

It's more urgent to create a new release, when clients require a new server version. But for me it is also not a problem sticking with "older" clients. e.g. in certain SW projects I wait at least 6 months before upgrading.

**Reply by tessus** [reply]
I haven't run any tests yet, but maybe you know. Will the new web client 2022.11.1 work with the current vw version 1.26.0? There could be fallback code in place.

**Reply by BlackDex** [reply]
There are some items not working, like export. And login is probably also going to be an issue of I remember.

**Comment by Franzz** [comment]
Thanks for the info about web client 2022.11.x not being compatible with vw 1.26. I was wondering the same thing. A compatibility matrix would be great though. Something like:

| Vaultwarden | Web Client |
| ------------- | ------------- |
|1.27 (i guess)|2022.11.x|
|1.26|2022.10.x|
|1.25.2|2022.6.2|
|1.25.0|2.28.1|
|1.23.1|2.25.0|
|1.23.0|2.23.0|
|1.22.2|2.21.1|

Just a thought.
